enumerate_methods.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+# This custom Enumerable module replicates the work of
+# ... the default Enumerable methods
+
+module Enumerable
+  def my_each
+    for i in (0...length)
+      yield(self[i])
+    end
+  end
+
+  def my_each_with_index
+    for i in (0...length)
+      yield(self[i], i)
+    end
+  end
+
+  def my_select
+    result = []
+    each do |current|
+      result.push(current) if yield(current)
+    end
+    return result
+  end
+
+  def my_all?
+    each do |current|
+      result = yield(current)
+      return false if result.nil? || result == false
+    end
+    return true
+  end
+
+  def my_none?
+    each do |current|
+      result = yield(current)
+      return false if result.nil? || result
+    end
+    return true
+  end
+
+  def my_count
+    counter = 0
+    each do |current|
+      result = yield(current)
+      counter += 1 if result == true
+    end
+    return counter
+  end
+
+  def my_map(proc = nil)
+    each do |current|
+      self[i] = yield(current)
```

Comment:
Where does the `i` come from?
you should use each with index in this case, and actually, you should use your own `my_each`, not the built-in each or `each_with_index`

Suggestion:
But really you should only need `my_each` to get each item and yield it and store the yielded result in another array, avoid manipulating self.

Reply:
This to allow the code review test to pass :)